### ISSUE TITLE
Data harvested as private

### ISSUE THREAD

**Issue by jbrown-xentity** [GSA/data.gov#3593]
Over 500K records in data.gov (40%+) are private, need to investigate if this is in error and make public what should be public.

Probably related to previous bug of defaulting harvest source to private.

## How to reproduce

1. Log into catalog (as admin)
1. Hit `https://catalog.data.gov/api/action/package_search?fq=(collection_package_id:* OR *:*)&include_private=true&rows=0` (or admin-catalog-next equivalent)

## Expected behavior

The above api call is similar to `https://catalog.data.gov/api/action/package_search?fq=(collection_package_id:* OR *:*)&rows=0`

## Actual behavior

Difference of ~500K

## Sketch

Probably will require evaluating harvest sources affected, manually changing datasets back to public, and re-indexing. Changing the harvest source flag will allow for new harvested data to be public, but will not change old datasets.

**Comment by FuhuXia** [comment]
This is related to https://github.com/GSA/data.gov/issues/897.
If a harvest source is (re)set to public, all existing private datasets should be set to public after a new harvest job. We can verify the issue and submit a bug report upstream. Once the bug is fixed, a re-harvest job will fix most of the datasets.

**Comment by jbrown-xentity** [comment]
I wonder if we should consider a change in harvest configuration to be "do a full reharvest", and that would be the bugfix/feature request upstream. Really any configuration change could result in different results, so it should be treated as such.
Unfortunately, I don't think the "gather" "fetch" "import" model really supports this "reset" or "full reharvest" option. It will require some research and thought, and may affect ckan harvest extensions in different ways.

**Comment by FuhuXia** [comment]
According the[ this spreadsheet](https://docs.google.com/spreadsheets/d/1_L8EMNGeLA_KU93DGXawre9kkoQJ4esOoFJo5hqAOjc), 50% existing datasets (642142/1291875) we harvested are affected and stay private.

**Comment by FuhuXia** [comment]
Other than the two approaches mentioned above, we can also develop a one-time custom script to call API and change all private datasets to public.

**Comment by FuhuXia** [comment]
Put all the pieces together
1. We fixed a previous [UI issue](https://github.com/GSA/data.gov/issues/897), but did not addresses existing private datasets caused by the issue. 
2. We tried to address the existing private dataset via DB query during the [cloud.gov migration](https://github.com/GSA/data.gov/issues/2788) but realized the query is too complicated to be practical.
3. Current `Dataset visibility` feature on harvest is not working. (#2984)
4. This also causes sitemap containing large amount of inaccessible private dataset. (#3879)
